client/valor/schemas/filters.py

```diff
@@ -1,8 +1,233 @@
+from typing import Any, Dict, List, Optional, Union
 
 from valor.enums import TaskType
-from valor.schemas.constraints import BinaryExpression, Constraint
+from valor.schemas.symbolic.atomics import (
+    Date,
+    DateTime,
+    Duration,
+    LineString,
+    MultiLineString,
+    MultiPoint,
+    MultiPolygon,
+    Point,
+    Polygon,
+    Time,
+    Variable,
+)
+from valor.schemas.symbolic.functions import (
+    And,
+    AppendableFunction,
+    Function,
+    Inside,
+    Intersects,
+    IsNotNull,
+    IsNull,
+    Negate,
+    OneArgumentFunction,
+    Or,
+    Outside,
+    TwoArgumentFunction,
+    Xor,
+)
+
+
+@dataclass
+class Constraint:
+    """
+    Represents a constraint with a value and an operator.
+
+    Attributes:
+        value : Any
+            The value associated with the constraint.
+        operator : str
+            The operator used to define the constraint.
+    """
+
+    value: Any
+    operator: str
+
+
+def _convert_symbol_to_attribute_name(symbol_name):
+    map_sym_to_attr = {
+        "dataset.name": "dataset_names",
+        "dataset.metadata": "dataset_metadata",
+        "model.name": "model_names",
+        "model.metadata": "model_metadata",
+        "datum.uid": "datum_uids",
+        "datum.metadata": "datum_metadata",
+        "annotation.task_type": "task_types",
+        "annotation.metadata": "annotation_metadata",
+        "annotation.box": "require_bounding_box",
+        "annotation.box.area": "bounding_box_area",
+        "annotation.polygon": "require_polygon",
+        "annotation.polygon.area": "polygon_area",
+        "annotation.raster": "require_raster",
+        "annotation.raster.area": "raster_area",
+        "annotation.labels": "labels",
+        "label.id": "label_ids",
+        "label.key": "label_keys",
+        "label.score": "label_scores",
+    }
+    return map_sym_to_attr[symbol_name]
+
+
+def _convert_expression_to_constraint(expr: Function):
+    # extract value
+    if isinstance(expr, TwoArgumentFunction):
+        variable = expr.rhs
+        if isinstance(
+            variable,
+            (
+                Point,
+                MultiPoint,
+                LineString,
+                MultiLineString,
+                Polygon,
+                MultiPolygon,
+            ),
+        ):
+            value = {
+                "type": type(variable).__name__.capitalize(),
+                "coordinates": variable.get_value(),
+            }
+        elif isinstance(variable, (DateTime, Date, Time, Duration)):
+            value = {type(variable).__name__.lower(): variable.encode_value()}
+        else:
+            value = variable.encode_value()
+    else:
+        value = None
+
+    # extract operator
+    if hasattr(expr, "_operator") and expr._operator is not None:
+        op = expr._operator
+    elif isinstance(expr, Inside):
+        op = "inside"
+    elif isinstance(expr, Intersects):
+        op = "intersect"
```

Comment:
nit: we use "intersect" here but "intersects" elsewhere

Reply:
This is for compatibility. Will be using `intersects` going forward.